docs/src/content/docs/guides/site-search.mdx

```diff
@@ -0,0 +1,167 @@
+---
+title: Site Search
+description: Learn about Starlight’s built-in site search features and how to customize them.
```

Comment:
Random thought: should we move the "New" badge from `guides/overriding-components.md` to this page?

Reply:
I think this page which contains valuable infos for **both** users of Pagefind and the first actual plugin may be way more interesting to people but it's your call ^^

Reply:
```suggestion
description: Learn about Starlight’s built-in site search features and how to customize them.
badge: New
```

Reply:
Accepted your suggestion and added the badge here for now! Maybe can move it to a plugin showcase in a bit once we have that.